## A/B Hypothesis Testing
The objective of Task 3 is to conduct A/B hypothesis testing. The goal is to test the following null hypotheses:

1. Hypothesis Testing Across Provinces.
2. Hypothesis Testing between Gender.

### **Load Libraries, and Dataset**

In [1]:
import os
import sys
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from scipy.stats import chi2_contingency

clean = pd.read_csv("../data/interim/cleaned_insurance_data.csv")


C:\Users\girma\AppData\Local\Temp\ipykernel_20596\2495312143.py:9: DtypeWarning: Columns (29) have mixed types. Specify dtype option on import or set low_memory=False.
  clean = pd.read_csv("../data/interim/cleaned_insurance_data.csv")


### **Hypothesis Testing: Across Providences**

For categorical risk comparisons (Claim Frequency):

✅ Chi-Squared Test

In [7]:
from scipy.stats import chi2_contingency

# Create contingency table for claims by province
contingency_table = pd.crosstab(clean['Province'], clean['TotalClaims'] > 0)
chi2_stat, p_value, _, _ = chi2_contingency(contingency_table)

print(f"Chi-Squared p-value: {p_value}")

Chi-Squared p-value: 5.925510718204678e-19


- Chi-Squared Statistic: A high test statistic suggests observed differences are unlikely due to random chance.
- p-value = 5.93e-19  → This is extremely low, far below the conventional threshold of 0.05.
- Conclusion: We strongly reject the null hypothesis (H₀: There are no risk differences across provinces).
- Business Impact: Provinces do exhibit significantly different claim frequencies, and ACIS should adjust its pricing & underwriting strategies accordingly.

One-Way ANOVA for Provinces



✅ F-Test(One way ANOVA)

In [8]:
import scipy.stats as stats

# Select relevant metric (e.g., Claim Severity)
province_groups = [clean.loc[clean['Province'] == p, 'TotalClaims'] for p in clean['Province'].unique()]

# Conduct one-way ANOVA test
anova_stat, p_value = stats.f_oneway(*province_groups)

print(f"ANOVA Test for Province-Based Claim Differences:\nStatistic: {anova_stat}, p-value: {p_value}")

ANOVA Test for Province-Based Claim Differences:
Statistic: 5.849413762407606, p-value: 1.6782057588675903e-07


This ANOVA test result tells us that there is a statistically significant difference in claim behavior across provinces.
Interpretation
- Statistic = 5.85 → Indicates that there is observable variation in claim amounts between provinces.
- p-value = 1.67e-07 (0.0000001678) → This is far below the threshold of 0.05, meaning we strongly reject the null hypothesis (H₀: There are no risk differences across provinces).
- What this means: At least one province has a significantly different claim severity or frequency compared to others.

### **Hypothesis Testing: Between Gender**

### **Hypothesis Statement**
- **Null Hypothesis (H₀):** There are no significant risk differences between Men and Women.
- **Alternative Hypothesis (H₁):** There **is** a statistically significant difference in claim risk based on gender.

### **Statistical Test Selection**
Since gender is **categorical**, we use:
- **Chi-Squared Test** for Claim Frequency (categorical)
- **T-Test** for Claim Severity (numeric)


In [10]:
# Create contingency table for claim frequency by gender
contingency_table = pd.crosstab(clean['Gender'], clean['TotalClaims'] > 0)
chi2_stat, p_value, _, _ = chi2_contingency(contingency_table)

print(f"Chi-Squared p-value: {p_value}")

Chi-Squared p-value: 0.003993781054238153



### **Interpretation of the Chi-Squared Test for Gender-Based Claim Frequency**
This test evaluates whether **men and women file claims at significantly different rates**.

#### **Key Results**
- **Chi-Squared Statistic:** Indicates observed differences in claim frequency across genders.
- **p-value = 0.00399** → This is **below the 0.05 threshold**, meaning the difference **is statistically significant**.
- **Conclusion:** We **reject the null hypothesis** (`H₀: There are no significant risk differences between Men and Women`).
- **Business Impact:** Gender **does** influence claim frequency, and ACIS might need to **review underwriting policies and pricing structures** accordingly.

### **2️⃣ Claim Severity Analysis (T-Test)**
This test checks whether **men and women have different claim amounts**.

In [11]:
from scipy.stats import ttest_ind

severity_men = clean.loc[clean['Gender'] == 'Male', 'TotalClaims']
severity_women = clean.loc[clean['Gender'] == 'Female', 'TotalClaims']

t_stat, p_value = ttest_ind(severity_men, severity_women, equal_var=False)
print(f"T-Test p-value: {p_value}")

T-Test p-value: 0.7669656471629474


### **Interpretation of the T-Test for Gender-Based Claim Severity**
This test evaluates **whether men and women have different average claim amounts** when a claim occurs.

#### **Key Results**
- **p-value = 0.76697** → **Much higher than the significance threshold (0.05).**
- **Conclusion:** We **fail to reject the null hypothesis** (`H₀: There are no significant risk differences between Men and Women in claim severity`).
- **Business Impact:** Gender **does not significantly affect** the severity of claims. Pricing and risk adjustments **should not differentiate based on gender alone**.